### PR TITLE
Allow bytestring-0.12

### PR DESCRIPTION
**Pull request by andreasabel**
Relax to `bytestring < 0.13` and test `bytestring-0.12` in CI via a `constraint-set`.

SQUASH ME!

### CODE REVIEW

File: cabal.haskell-ci [hunk 1]
```diff
@@ -2,6 +2,6 @@ branches: master ci*
 
 -- unconstrained: False
 
-constraint-set bytestring-0.11
-  ghc: == 8.10.7
-  constraints: bytestring >=0.11 && <0.12
+constraint-set bytestring-0.12
+  ghc: >= 9.6
+  constraints: bytestring >=0.12
```
Comment (andreasabel) [inline cabal.haskell-ci:7]: This `constraint-set` checks that `base64-bytestring` builds with `bytestring-0.12` on CI.
However, it seems that Haskell CI does not run the tests again for a `constraint-set` build.
I ran the tests locally.

**Comment by andreasabel** [comment]
In https://github.com/haskell/base64-bytestring/actions/runs/5505398540/jobs/10032701315 I attempt to also build the tests in `constraint-set bytestring-0.12`, but this fails because `text` does not allow `bytestring-0.12` yet and `constraint-set` has no `allow-newer` option that would allow `text` to build with `bytestring-0.12`.

Re:
- https://github.com/haskell-hvr/regex-tdfa/pull/54#issuecomment-1627609263

**Comment by andreasabel** [comment]
Following @phadej's advice I added a global `allow-newer: bytestring` to CI.  This is ignored in the usual runs because `bytestring` has a `constraint installed` there.  So, it should only kick in in the `constraint-set bytestring-0.12`.
This makes the building of the test suite succeed with `bytestring-0.12`.  However, I am a bit puzzled why it does not show up in the list of packages that cabal builds in this step: https://github.com/haskell/base64-bytestring/actions/runs/5505912132/jobs/10033887626?pr=50#step:25:24

**Comment by phadej** [comment]
>  However, I am a bit puzzled why it does not show up in the list of packages that cabal builds in this step

Probably because `bytestring-0.12` already in the store (from previous runs).

I guess it won't hurt to add `cabal-plan topo` to the constraint set builds.

**Review by emilypi — APPROVED** [review]
Approving to move this along, but I want to move away from haskell-ci and yet another undocumented tool, so don't get your hopes up about investing too much into the current CI setup.

**Comment by andreasabel** [comment]
Thanks, @emilypi!
Can you please facilitate this `bytestring` bump in a hackage revision?

**Comment by emilypi** [comment]
It builds on master, so sure

**Comment by andreasabel** [comment]
https://hackage.haskell.org/package/base64-bytestring-1.2.1.0/revisions/

**Pull request merged**